tests/ga/test_update.py

```diff
@@ -1740,6 +1741,66 @@ def test_it_should_recreate_extension_event_directories_for_existing_extensions_
                 for ext_dir in expected_events_dirs:
                     self.assertTrue(os.path.exists(ext_dir), "Extension directory {0} should exist!".format(ext_dir))
 
+    def test_it_should_report_update_status_in_status_blob(self):
+        with _get_update_handler(iterations=1) as (update_handler, protocol):
+            with patch.object(conf, "get_autoupdate_gafamily", return_value="Prod"):
+                with patch("azurelinuxagent.common.logger.warn") as patch_warn:
+
+                    protocol.aggregate_status = None
+                    protocol.incarnation = 1
+
+                    def mock_http_put(url, *args, **_):
+                        if HttpRequestPredicates.is_host_plugin_status_request(url):
+                            # Skip reading the HostGA request data as its encoded
+                            return MockHttpResponse(status=500)
+                        protocol.aggregate_status = json.loads(args[0])
+                        return MockHttpResponse(status=201)
+
+                    def update_goal_state_and_run_handler():
+                        protocol.incarnation += 1
+                        protocol.mock_wire_data.set_incarnation(protocol.incarnation)
+                        self._add_write_permission_to_goal_state_files()
+                        update_handler.set_iterations(1)
+                        update_handler.run(debug=True)
+                        self.assertTrue(update_handler.exit_mock.called, "The process should have exited")
+                        exit_args, _ = update_handler.exit_mock.call_args
+                        self.assertEqual(exit_args[0], 0,
+                                         "Exit code should be 0; List of all warnings logged by the agent: {0}".format(
+                                             patch_warn.call_args_list))
+
+                    protocol.set_http_handlers(http_put_handler=mock_http_put)
+
+                    # Case 1: No requested version in GS; updateStatus should not be reported
+                    update_handler.run(debug=True)
+                    self.assertFalse("updateStatus" in protocol.aggregate_status['aggregateStatus']['guestAgentStatus'],
+                                     "updateStatus should not be reported if not asked in GS")
+
+                    # Case 2: Requested version in GS != Current Version; updateStatus should not be error
```

Comment:
this comment seems backwards... Case 2 is expecting an error, right?
```python
self.assertEqual(VMAgentUpdateStatuses.Error, update_status['status'], "Status should be an error")
```

Reply:
Good catch! I think I got confused between the wording of case 2 and 3. Fixed it thanks!